ci/generate-and-upload-build-steps.sh

```diff
@@ -1,15 +1,73 @@
 #!/bin/bash
 set -euo pipefail
 
+upload_build_configuration_step() {
+    cat "ci/gdk_build.template.steps.yaml" | \
+    sed "s|ENGINE_COMMIT_HASH_PLACEHOLDER|$1|g" | \
+    sed "s|BUILD_PLATFORM_PLACEHOLDER|$2|g" | \
+    sed "s|BUILD_TARGET_PLACEHOLDER|$3|g" | \
+    sed "s|BUILD_STATE_PLACEHOLDER|$4|g" | \
+    buildkite-agent pipeline upload
+}
+
+generate_build_configuration_steps () {
+    # See https://docs.unrealengine.com/en-US/Programming/Development/BuildConfigurations/index.html for possible configurations 
+
+    # if BUILD_ALL_CONFIGURATIONS environment variable exists AND is equal to "true", then...
+    if [[ "${BUILD_ALL_CONFIGURATIONS:-false}" = "true" ]]; then
+        echo "This is a nightly build. Generating the appropriate steps..."
+        
+        # Editor builds (Test and Shipping build states do not exist for the Editor build target)
+        for build_state in "DebugGame" "Development"; do
+             upload_build_configuration_step "$1" "Win64" "Editor" "$build_state"
```

Comment:
What is `$1`? Its generally good practice (imo) to resolve parameters to the script to named arguments at the start of the script.

I.e. - 

```
DESCRIPTIVE_NAME="${1}"
```

Reply:
Fixed!